rust/noosphere-gateway/src/handlers/v0alpha1/push.rs

```diff
@@ -325,7 +317,7 @@ where
     async fn notify_name_resolver(&self) -> Result<()> {
         if let Some(name_record) = &self.request_body.name_record {
             if let Err(error) = self.name_system_tx.send(NameSystemJob::Publish {
-                context: self.sphere_context.clone(),
+                context: self.gateway_sphere.to_inner(),
```

Comment:
Why isn't it sufficient to pass the `GatewaySphereContext` directly here? That is to ask: why `to_inner`?

Reply:
`C` is always the inner sphere context -- `Arc<Mutex<SphereContext<S>>>` I believe. Returning the `gateway_sphere` is a different newtype. Unfortunately, newtype is required since we can't have an extractor for a generic `C` type

Reply:
This just begs the question: why can't `C` be the `GatewaySphereContext` type?

Reply:
Documenting different approaches here on matching the `C` type with the arguments used in routes:

* Approach 1: `C: HasMutableSphereContext<S> + FromRequestParts<_, _> `: we can't generally implement FRP for C (S not trait bound), having the consumer implement is fraught due to the trait being dependent on axum state/middleware. We could implement for concrete types of C maybe (`Arc<Mutex<SphereContext<S>>>`), but then our top level generics aren't that generic. thinking more, i think we explicitly do want to implement FRP over a newtype, as its what interacts with the gateway manager, all specific to our other extractors
* Approach 2: `Gateway::new::<M, GatewaySphereContext<Arc<Mutex<SphereContext<S>>>>, S>(..)`  we could expose our newtype wrapper, with its own HasMutableSphereContext traits, but seems irrelevant on the consumer side
* Approach 3: `sphere_extractor: SphereExtractor<C, S>` implement FRP over a C wrapper and immediately convert to C; not elegant though not much different from Arc<T> types

opting for 3 here, as for now we explicitly want to define the FRP used in routes, and while it'd be nice for routes to have `route_handler(context: C)`, introducing a newtype only for axum parsing seems reasonable